pandas/core/resample.py

```
@@ -2763,7 +2768,10 @@ def _adjust_dates_anchored(
     if offset is not None:
         offset = offset.as_unit(unit)
 
-    freq_value = Timedelta(freq).as_unit(unit)._value
+    if isinstance(freq, Day):
```

In https://github.com/pandas-dev/pandas/issues/41943 I was hoping that if `D` means calendar day, we wouldn't need to force to be a fixed frequency in resample

---

yah i was hoping for this too but wasn't able to get it working.  will give it another try after addressing other comments

---

Based on test_resample_tz_localized2:

```
idx = date_range("2001-09-20 15:59", "2001-09-20 16:00", freq="min", tz="Australia/Sydney")
s = Series([1, 2], index=idx)
    
result = s.resample("D", closed="right", label="right").mean()
ex_index = date_range("2001-09-21", periods=1, freq="D", tz="Australia/Sydney")
expected = Series([1.5], index=ex_index)
    
tm.assert_series_equal(result, expected)  # Fails with actual result being...

ex_index = pd.date_range("2001-09-20", "2001-09-21", tz="Australia/Sydney", freq="D")
expected = Series([np.nan, 1.5], index=ex_index)
```

---

The test_resample_origin_with_day_freq_on_dst case is a bit harder bc it uses the "offset" keyword

```
# test complex behavior of origin/offset in a DST context
tz = "America/Chicago"
unit = "ns"
start = Timestamp("2013-11-03", tz=tz)
end = Timestamp("2013-11-03 23:59", tz=tz)
rng = date_range(start, end, freq="1h").as_unit(unit)
ts = Series(np.ones(len(rng)), index=rng)

result = ts.resample("D", origin="start", offset="-2h").sum()
ex_index = DatetimeIndex(["2013-11-02 22:00-05:00", "2013-11-03 22:00-06:00"], tz=tz, freq="D")
expected = Series([23.0, 2.0], index=ex_index)
tm.assert_series_equal(result, expected)
```

---

Maybe @jorisvandenbossche will have an intuition for if these make results make sense?

---

sounds good.  ill update the expecteds.

---

Looking at the NDFrame.resample docstring, offset is added to origin, and origin only takes effect for Tick-frequencies.  I'll do a follow up to warn if those params are passed and silently ignored.